### ISSUE TITLE
Training with AMP gives overflow warning

### ISSUE THREAD

**Issue by zshn25** [pytorch/pytorch#42460]
## 🐛 Bug

WARNING:root:NaN or Inf found in input tensor while using AMP but no such warning without AMP

AMP gives the following warning

```
/lib/python3.7/site-packages/numpy/core/fromnumeric.py:90: RuntimeWarning: overflow encountered in reduce
  return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
WARNING:root:NaN or Inf found in input tensor.
``
Sorry, couldn't reproduce it outside my codebase.

## Expected behavior

No warnings

## Environment

 - PyTorch Version (e.g., 1.0): 1.6.0
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source): conda install pytorch torchvision cudatoolkit=10.2 -c pytorch
 - Python version: 3.7
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: DistributedDataParallel with one process per GPU on single machine
 - Any other relevant information:

## Additional context

<!-- Add any other context about the problem here. -->

**Comment by aqrliu** [comment]
get the same problem as yours
